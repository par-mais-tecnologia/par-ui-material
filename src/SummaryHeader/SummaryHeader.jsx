import React, { PureComponent } from 'react'

class SummaryHeader extends PureComponent {
  getStyle (headerProps) {
    return {
      ...headerProps,
      height: headerProps && headerProps.heigth ? headerProps.heigth : 'auto',
      backgroundImage: headerProps && headerProps.backgroundImage ? headerProps.backgroundImage : 'linear-gradient(26deg, rgb(59,133,135), rgb(62,167,172))'
    }
  }

  render () {
    const { classes, headerProps } = this.props

    return (
      <div className={`flex items-center w-100 ${classes}`} style={this.getStyle(headerProps)}>
        {this.props.children}
      </div>
    )
  }
}

export default SummaryHeader
