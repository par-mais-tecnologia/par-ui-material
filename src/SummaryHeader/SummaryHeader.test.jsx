import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import SummaryHeader from './'

configure({ adapter: new Adapter() })

export const test = describe('SummaryHeader', () => {
  it('shallow Header', () => {
    const wrapper = shallow(
      <SummaryHeader />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow Header match Snapshot', () => {
    const wrapper = shallow(
      <SummaryHeader />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
