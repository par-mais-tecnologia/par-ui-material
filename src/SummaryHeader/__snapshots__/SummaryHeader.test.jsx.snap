// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryHeader shallow Header match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SummaryHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "flex items-center w-100 undefined",
      "style": Object {
        "backgroundImage": "linear-gradient(26deg, rgb(59,133,135), rgb(62,167,172))",
        "height": "auto",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "flex items-center w-100 undefined",
        "style": Object {
          "backgroundImage": "linear-gradient(26deg, rgb(59,133,135), rgb(62,167,172))",
          "height": "auto",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
